database/service_db_test.go

```diff
@@ -60,22 +76,68 @@ func TestServiceDBDelete(t *testing.T) {
 	db, closer := openServiceDB(t)
 	defer closer()
 
-	s := &service.Service{ID: "1", Name: "test-service"}
+	// id
+	s := &service.Service{ID: "00", Alias: "2", Name: "test-service"}
 	require.NoError(t, db.Save(s))
 	require.NoError(t, db.Delete(s.ID))
 	_, err := db.Get(s.ID)
-	require.True(t, IsErrNotFound(err))
+	require.IsType(t, &ErrNotFound{}, err)
+	_, err = db.Get(s.Alias)
+	require.IsType(t, &ErrNotFound{}, err)
+
+	_, err = db.db.Get([]byte(idKeyPrefix+s.ID), nil)
+	require.Equal(t, leveldb.ErrNotFound, err)
+	_, err = db.db.Get([]byte(aliasKeyPrefix+s.Alias), nil)
+	require.Equal(t, leveldb.ErrNotFound, err)
+
+	// alias
+	s = &service.Service{ID: "11", Alias: "3", Name: "test-service"}
+	require.NoError(t, db.Save(s))
+	require.NoError(t, db.Delete(s.Alias))
+	_, err = db.Get(s.Alias)
+	require.IsType(t, &ErrNotFound{}, err)
+	_, err = db.Get(s.ID)
+	require.IsType(t, &ErrNotFound{}, err)
+
+	_, err = db.db.Get([]byte(idKeyPrefix+s.ID), nil)
+	require.Equal(t, leveldb.ErrNotFound, err)
+	_, err = db.db.Get([]byte(aliasKeyPrefix+s.Alias), nil)
+	require.Equal(t, leveldb.ErrNotFound, err)
+}
+
+func TestServiceDBDeleteConcurrency(t *testing.T) {
```

Comment:
I add this test in order to make sure that Delete function behave correctly on concurrent deletion.
If you remove the transaction in Delete function, then this test fails.

To test, replace in Delete function:
```
tx, err := d.db.OpenTransaction()
```
By
```
tx = db.db.
```
And remove `tx.Discard()` and `tx.Commit()`